### PR TITLE
feat: simplify dependencies

### PR DESCRIPTION
**Pull request by pommedeterresautee**
* simplify dependencies
* improve some comments regarding matmul trick

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -31,7 +31,7 @@ RUN update-alternatives --install /usr/bin/python python /usr/bin/python3.10 1 &
 RUN python3.9 -m ensurepip --default-pip --upgrade && \
     pip install --upgrade pip
 
-RUN pip install --pre torch==2.0.0 --extra-index-url https://download.pytorch.org/whl/cu118
+RUN pip install --pre torch==2.0.0
```
Comment (gaetansnl) [inline Dockerfile:34]: do we still need to have it here ?

File: src/kernl/implementations/linear_layer.py [hunk 2]
```diff
@@ -155,7 +155,14 @@ def kernel_fma(
     m_offs_untagged = block_m_idx * BLOCK_M + tl.arange(0, BLOCK_M)
     n_offs_untagged = block_n_idx * BLOCK_N + tl.arange(0, BLOCK_N)
 
-    # trick to avoid masking on M and N axis byt tagging variables
+    # trick to avoid masking on M and N axis
+    # rm and rn can contains addresses outside matrix boundaries
```
Comment (gaetansnl) [inline src/kernl/implementations/linear_layer.py:159]: names are wrong I think

File: src/kernl/implementations/linear_layer.py [hunk 3]
```diff
@@ -155,7 +155,14 @@ def kernel_fma(
...
+    # modulo operation is used to wrap around the indices that go beyond the matrix boundaries
+    # The value loaded are not ok but at least we are not reading outside the A/B matrices
+    # Then, during storing in C a mask is used and the results related to these wrong values is discarded!
+    # Regarding max_contiguous and multiple_of, they are used to force the compiler to vectorize loads
```
Comment (gaetansnl) [inline src/kernl/implementations/linear_layer.py:163]: "force the compiler to vectorize loads" reference ?

File: src/kernl/implementations/linear_layer.py [hunk 4]
```diff
@@ -155,7 +155,14 @@ def kernel_fma(
...
+    # multiple_of indicates that the first element of rm / rn is a multiple of BLOCK_M
```
Comment (gaetansnl) [inline src/kernl/implementations/linear_layer.py:164]: missing BLOCK_M / BLOCK_N, same below

**Review by gaetansnl — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (pommedeterresautee) [reply Dockerfile:34]: it s a huge dependency, having it here avoid rebuilding most of the image for each change of code?

File: src/kernl/implementations/linear_layer.py (hunk 4 above)
Comment (pommedeterresautee) [reply src/kernl/implementations/linear_layer.py:164]: done

File: src/kernl/implementations/linear_layer.py (hunk 2 above)
Comment (pommedeterresautee) [reply src/kernl/implementations/linear_layer.py:159]: oups got them from the official matmul

**Pull request merged**